File: codegen/results/v002/qwen3.6_35b_think/20260418_180142/python/001_csv_to_json/solution.py
import csv
import json

def main():
    data = []
    with open('input/data.csv', mode='r', encoding='utf-8') as f:
        reader = csv.DictReader(f)
        for row in reader:
            data.append({
                'Name': row['Name'].strip(),
                'Age': int(row['Age'].strip()),
                'Email': row['Email'].strip(),
                'Score': float(row['Score'].strip())
            })
    print(json.dumps(data))

if __name__ == '__main__':
    main()

File: codegen/results/v002/qwen3.6_35b_think/20260418_180142/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"log"
	"os"
	"strconv"
)

type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	f, err := os.Open("input/data.csv")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	reader := csv.NewReader(f)
	records, err := reader.ReadAll()
	if err != nil {
		log.Fatal(err)
	}

	var people []Person
	for _, row := range records[1:] {
		if len(row) < 4 {
			continue
		}
		age, err := strconv.Atoi(row[1])
		if err != nil {
			log.Fatal(err)
		}
		score, err := strconv.ParseFloat(row[3], 64)
		if err != nil {
			log.Fatal(err)
		}
		people = append(people, Person{
			Name:  row[0],
			Age:   age,
			Email: row[2],
			Score: score,
		})
	}

	jsonData, err := json.Marshal(people)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(jsonData))
}